app/controllers/message_handler_controller.rb

```diff
@@ -0,0 +1,5 @@
+class MessageHandlerController < ApplicationController
+  def handle
+
```

Comment:
I'm leaving that section intentionally blank at the moment because the necessary code to make that work is going to balloon the PR pretty significantly, this PR was already large enough, and could be pushed up without affecting anything else at the moment.